vm_insnhelper.c

```diff
@@ -2524,6 +2553,21 @@ CALLER_SETUP_ARG(struct rb_control_frame_struct *restrict cfp,
              */
             vm_caller_setup_arg_kw(cfp, calling, ci);
         }
+        else if (UNLIKELY(RB_CFUNC_HEAP_ARGV_P(calling))) {
```

Comment:
`if (allow_heap_argv && calling->heap_argv)` is better because a C compiler can detect that this part is removable on many cases.